### DISCUSSION TITLE
JS Twig tag - add param for wrapping code in self executing function

### DISCUSSION THREAD

**Discussion by piotrpog** [craftcms/cms#9049] [Ideas]
I usually wrap my jsvascript inside `{% js %}` wiyth self executing function to avoid collision. It would be useful if there was some additional param for js tag for that.

**Comment by brandonkelly** [comment]
Good idea. Just added this for Craft 3.7.

**Comment by piotrpog** [comment]
@brandonkelly 
Thanks! But could you also add option to overwrite that? Sometimes i need to NOT wrap this code in self executing function, for example when i use Alpine JS.
I think that ideally wrapping in self executing function would be optional to ensure old code wont break.

**Comment by brandonkelly** [comment]
Not sure I’m following… how would this break Alpine?

**Comment by piotrpog** [comment]
I dont know why exactly iyt happened, but it seems that alpine js object needed to be defined globally, so that it gets attached to HTML. When i wrapped alpine js object inside self executing function, it stopped working.

**Comment by brandonkelly** [comment]
Ah of course 🤦🏻  any JS code that assumes a global scope and does `someVar = …` rather than `window.someVar = …` would break.

Jut reverted the change for now, and will reconsider for 4.0. I think it’s a good idea and I don’t think it should be configurable. It’s reasonable to expect that all JS code is isolated, and must explicitly include `window.` if it means to define a variable in the global scope.
